packages/authentication/src/strategy-adapter.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import * as http from 'http';
+import {ServerRequest as Request} from 'http';
+import {ParsedRequest} from '@loopback/core';
+
+/**
+ * Interface definition of a passport strategy.
+ */
+export interface Strategy {
+  authenticate(req: http.ServerRequest): void;
+}
+
+/**
+ * Shimmed Request to satisfy express requirements of passport strategies.
+ */
+export class ShimRequest {
```

Comment:
Fair enough. Let's keep the current design then, we can easily improve it later if there is a need for that.